python/ray/tests/test_runtime_env_failure.py

```diff
@@ -0,0 +1,155 @@
+import os
+from unittest import mock
+
+import pytest
+from ray._private.runtime_env.packaging import (
+    RAY_RUNTIME_ENV_FAIL_DOWNLOAD_FOR_TESTING_ENV_VAR,
+    RAY_RUNTIME_ENV_FAIL_UPLOAD_FOR_TESTING_ENV_VAR,
+)
+import ray
+
+from ray.runtime_env import RuntimeEnv
+
+
+def using_ray_client(address):
+    return address.startswith("ray://")
+
+
+@pytest.fixture
+def set_agent_failure_env_var():
+    os.environ["_RAY_AGENT_FAILING"] = "1"
+    yield
+    del os.environ["_RAY_AGENT_FAILING"]
+
+
+# TODO(SongGuyang): Fail the agent which is in different node from driver.
+@pytest.mark.skip(
+    reason="Agent failure will lead to raylet failure and driver failure."
+)
+@pytest.mark.parametrize("runtime_env_class", [dict, RuntimeEnv])
+def test_runtime_env_broken(
```

Comment:
Is this the correct name of the test? why is `env_vars={"TF_WARNINGS": "none"}` invalid?